web/src/app/services/data-store/data-store.service.ts

```diff
@@ -312,34 +313,68 @@ export class DataStoreService {
   }
 
   lois$({id}: Survey, {email}: User): Observable<List<LocationOfInterest>> {
-    const predefinedLois = this.db.collection(
-      `${SURVEYS_COLLECTION_NAME}/${id}/lois`,
-      ref => ref.where('predefined', '==', true)
-    );
-
-    const userLois = this.db.collection(
-      `${SURVEYS_COLLECTION_NAME}/${id}/lois`,
-      ref =>
+    return this.db
+      .collection(SURVEYS_COLLECTION_NAME, ref =>
         ref
-          .where('predefined', '==', false)
-          .where('created.user.email', '==', email)
-    );
-
-    return combineLatest([
-      predefinedLois.valueChanges({idField: 'id'}),
-      userLois.valueChanges({idField: 'id'}),
-    ]).pipe(
-      map(lois => {
-        const [predefinedLois, userLois] = lois;
-        const combined = predefinedLois.concat(userLois);
-        return List(
-          combined
-            .map(obj => LoiDataConverter.toLocationOfInterest(obj.id, obj))
-            .filter(DataStoreService.filterAndLogError<LocationOfInterest>)
-            .map(loi => loi as LocationOfInterest)
-        );
-      })
-    );
+          .where(documentId(), '==', id)
+          .where(new FieldPath('acl', email), '==', Role.SURVEY_ORGANIZER)
+      )
+      .get()
+      .pipe(
+        switchMap(({empty: userIsNotSurveyOrganizer}) => {
+          if (userIsNotSurveyOrganizer) {
+            const predefinedLois = this.db.collection(
+              `${SURVEYS_COLLECTION_NAME}/${id}/lois`,
+              ref => ref.where('predefined', 'in', [true, null])
+            );
+
+            const userLois = this.db.collection(
+              `${SURVEYS_COLLECTION_NAME}/${id}/lois`,
+              ref =>
+                ref
+                  .where('predefined', '==', false)
+                  .where('created.user.email', '==', email)
+            );
+
+            return combineLatest([
+              predefinedLois.valueChanges({idField: 'id'}),
+              userLois.valueChanges({idField: 'id'}),
+            ]).pipe(
+              map(([predefinedLois, userLois]) => {
+                const combined = predefinedLois.concat(userLois);
+                return List(
+                  combined
```

Comment:
Sorry I have a short memory - to remember what is combined here. Let's just call it something like `accessibleLois`, and rename the method to match?

Also, can we update this method to just take `surveyId` and `userEmail`? (principle of least privelege)